src/data/int/basic.lean

```diff
@@ -1046,6 +1046,29 @@ end
 
 @[simp] theorem neg_add_neg (m n : ℕ) : -[1+m] + -[1+n] = -[1+nat.succ(m+n)] := rfl
 
+lemma dvd_linear {d x y a b : ℤ} (hdx : d ∣ x) (hdy : d ∣ y) :  d ∣ (a*x + b*y) :=
```

Comment:
You can easily prove it for any semiring with `dvd_add (hdx.mul_left a) (hdy.mul_left b)`.

Reply:
This one-liner works for `ℤ`, but if I replace `ℤ` with `{α : Type*} [semiring α]` I get a `type mismatch` error: `mul_left` is expecting an argument of the form `?m_3 ∣ ?m_4` which `hdx` somehow doesn't match.

Reply:
`mul_left` needs commutativity, unlike `mul_right`; proof seems to go through with `{α : Type*} [comm_semiring α]`

Reply:
Thanks!  I've proved the more general lemma and moved it to `algebra/ring/basic`.